### PR TITLE
fix: channel match

### PR DESCRIPTION
**Pull request by nuanyang233**
# Pull Request

## Related Issue

Closes FW-1082

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/components/Providers.tsx [hunk 1]
```diff
@@ -91,7 +96,7 @@ export function Providers(props: { children: React.ReactNode }) {
     return (
         <I18nProvider i18n={i18n}>
             {/* We are using @tanstack/react-query@5.8.7 */}
-            <QueryClientProvider client={queryClient}>
+            <PersistQueryClientProvider client={queryClient} persistOptions={{ persister }}>
```
Comment (guanbinrui) [inline src/components/Providers.tsx:99]:
```suggestion
            <PersistQueryClientProvider client={queryClient} persistOptions={persistOptions}>
```

File: src/store/useChannelStore.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    persist(
+        immer((set, get) => ({
+            allChannelData: {
+                [Source.Farcaster]: {},
+                [Source.Lens]: {},
+                [Source.Twitter]: {},
+            },
+            addChannel: (source: SocialSource, key: string, channel: Channel | null) => {
+                set((state) => {
+                    state.allChannelData[source][key] = channel;
```
Comment (guanbinrui) [inline src/store/useChannelStore.ts:24]:
```suggestion
                    if (channel) {
                        state.allChannelData[source][key] = channel;
                    }
```

Comment (nuanyang233) [reply src/store/useChannelStore.ts:24]: No, the channel can be null.

File: src/components/Markup/MarkupLink/ChannelTag.tsx [hunk 3]
```diff
@@ -31,11 +33,23 @@ export const ChannelTag = memo<Omit<MarkupLinkProps, 'post'>>(function ChannelTa
...
+        queryKey: ['channel', 'tag', source, channelId],
         queryFn: async () => {
             if (!channelId || !source) return;
-            const provider = resolveSocialMediaProvider(source);
-            return provider.getChannelById(channelId);
+            const cache = allChannelData[source][channelId];
+            // If the cache be null, the channel handle does not exist.
+            if (cache === null) return;
+
+            // If the cache be undefined, the query is first query.
```
Comment (guanbinrui) [inline src/components/Markup/MarkupLink/ChannelTag.tsx:43]:
```suggestion
            // if the cache is undefined, it indicates the first query.
```

**Pull request merged**